core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1138,6 +1185,7 @@ private void acquireSubsetBatchRecords(
         } finally {
             lock.writeLock().unlock();
         }
+        return result.size();
```

Comment:
Apologies, but why don't we return the count of AcquiredRecords added to the result in this method? Since result is reused by the caller, result.size won't equal the count of the subset, correct?

Reply:
> Currently in this method we add result for every offset hence the count is the size of the result

We're using the same results object when calling `acquireSubsetBatchRecords`, so results.size will return the total count of all added offsets across each call. Does the condition `if (!fullMatch || inFlightBatch.offsetState() != null)` get matched only once while looping through `subMap`?
```
                    int acquiredSubsetCount = acquireSubsetBatchRecords(memberId, firstBatch.baseOffset(), lastBatch.lastOffset(), inFlightBatch, result);
                    acquiredCount += acquiredSubsetCount;
```

Reply:
You are so right, my bad. I have fixed this. While working further on this I will change some existing subset tests to account for these cases as well. Thankf for catching, I have pushed the change.

Reply:
@chia7712 Please let me know if we there is any other feedback to address, or we are good to go?